packages/auth/src/app.ts

```diff
@@ -342,12 +344,29 @@ export class App {
 
     koa.use(cors())
     koa.keys = [this.config.cookieKey]
+
+    const redis = await this.container.use('redis')
+    const maxAgeMs = 60 * 1000
     koa.use(
       session(
         {
           key: 'sessionId',
-          maxAge: 60 * 1000,
-          signed: true
+          maxAge: maxAgeMs,
+          signed: true,
+          store: {
+            async get(key) {
+              const data = await redis.get(key)
+              return data ? JSON.parse(data) : null
+            },
+            async set(key, session) {
+              // Add a delay to cookie age to ensure redis record expires after cookie
+              const expireInMs = maxAgeMs + 10 * 1000
+              await redis.psetex(key, expireInMs, JSON.stringify(session))
+            },
```

Comment:
```suggestion
              await redis.set(key, JSON.stringify(session), 'PX', expireInMs)
```
I think `psetex` is deprecated based on the docs

Reply:
`psetex` is not deprecated per my knowledge, but it is not wise to use it, since it will be deprecated and removed.
From the docs:
```
Note: Since the SET command options can replace SETNX, SETEX, PSETEX, GETSET, it is possible that in future versions of Redis these commands will be deprecated and finally removed.
```

Reply:
BTW, no need to do stringify on session object. Instead, you can just use hmset(key, session)

Reply:
Updated to this ^ except `hset`. apparently `hmset`, like `psetex` is marked as deprecated in `redis` docs https://redis.io/docs/latest/commands/hmset/

[dda0079](https://github.com/interledger/rafiki/pull/2693/commits/dda0079fafeb6f6e8da4206e3e90ee94f556c839)